hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala

```diff
@@ -0,0 +1,352 @@
+package org.broadinstitute.monster.dap.dog
+
+import org.broadinstitute.monster.dap.RawRecord
+import org.broadinstitute.monster.dogaging.jadeschema.fragment.HlesDogResidentialEnvironment
+
+object ResidentialEnvironmentTransformations {
+
+  /** Map all residential environment fields out of a raw RedCap record into a partial Dog model. */
+  def mapResidentialEnvironment(rawRecord: RawRecord): HlesDogResidentialEnvironment = {
+    val init = HlesDogResidentialEnvironment.init()
+
+    val transformations = List(
+      mapPastResidences _,
+      mapHouse _,
+      mapHeating _,
+      mapDrinkingWater _,
+      mapToxinExposure _,
+      mapFlooring _,
+      mapProperty _,
+      mapNeighborhood _
+    )
+
+    transformations.foldLeft(init)((acc, f) => f(rawRecord, acc))
+  }
+
+  /**
+    * Parse all past-residence-related fields out of a raw RedCap record,
+    * injecting them into a partially-modeled dog record.
+    */
+  def mapPastResidences(
+    rawRecord: RawRecord,
+    dog: HlesDogResidentialEnvironment
+  ): HlesDogResidentialEnvironment = {
+    val secondaryHome = rawRecord.getOptionalBoolean("oc_address2_yn")
+    val pastResidenceCount = rawRecord.getOptionalNumber("de_home_nbr")
+    val currentResidenceCount = secondaryHome.flatMap {
+      case true  => Some(2L)
+      case false => Some(1L)
+    }
+    if (pastResidenceCount.isDefined) {
+      val totalResidenceCount = pastResidenceCount.head + currentResidenceCount.getOrElse(0L)
```

Comment:
`.isDefined` + `.head` not fully purged!

I think this would be equivalent:
```scala
pastResidenceCount.fold(dog.copy(totalResidenceCount = Some(currentResidenceCount))) { pastCount =>
  // Use 'pastCount' where you have `.head` calls now
}
```